### ISSUE TITLE
Namespace is required by ClusterRoleBindings

### ISSUE THREAD

**Issue by sudermanjr** [FairwindsOps/rbac-manager#86]
It was reported that ClusterRoleBindings currently require namespaces. This should not be the case.

@thejosephstevens, could you please post the rbacdefinition and logging associated with this?

**Comment by thejosephstevens** [comment]
Happy to! The log is: (I can get you more log context if that would be helpful)
```
time="2019-09-29T13:00:51Z" level=info msg="Reconciling RBACDefinition node-labeler"
time="2019-09-29T13:00:51Z" level=info msg="Creating Service Account: node-labeler"
time="2019-09-29T13:00:51Z" level=error msg="Error creating Service Account: an empty namespace may not be set during creation"
time="2019-09-29T13:00:52Z" level=info msg="Creating Cluster Role Binding: node-labeler-node-labeler-node-labeler"
time="2019-09-29T13:00:52Z" level=error msg="Error creating Cluster Role Binding: ClusterRoleBinding.rbac.authorization.k8s.io \"node-labeler-node-labeler-node-labeler\" is invalid: subjects[0].namespace: Required value"
```
and the rbacdefinition is:
```
apiVersion: rbacmanager.reactiveops.io/v1beta1
kind: RBACDefinition
metadata:
  name: node-labeler
rbacBindings:
  - name: node-labeler
    subjects:
      - kind: ServiceAccount
        name: node-labeler
    clusterRoleBindings:
      - namespace: infra
        clusterRole: node-labeler
```

I'm pretty sure I tried adding the namespace to the subject and not the clusterrolebinding, but I'm having a hard time finding the logs from that state. I can recreate it if that would be helpful.

**Comment by robscott** [comment]
Hey @thejosephstevens, thanks for the detailed report! In this case you need to specify the namespace the ServiceAccount is in since the same name could exist in multiple namespaces. (This isn't really an RBAC Manager thing, more just a Kubernetes RBAC thing). So something like this should work:
```
apiVersion: rbacmanager.reactiveops.io/v1beta1
kind: RBACDefinition
metadata:
  name: node-labeler
rbacBindings:
  - name: node-labeler
    subjects:
      - kind: ServiceAccount
        name: node-labeler
        namespace: your-namespace-goes-here
    clusterRoleBindings:
      - namespace: infra
        clusterRole: node-labeler
```

**Comment by robscott** [comment]
I should add that RBAC Manager really should be validating this on it's input instead of just passing things through without validation as is happening here. Unfortunately I don't _think_ CRD schema validation can catch this, so it would still end up being something that only surfaced in RBAC Manager logs, but at least RBAC Manager wouldn't try to create an invalid Cluster Role Binding.

**Comment by thejosephstevens** [comment]
I thought I actually tried that particular config and it failed, but let me attempt to repro so we can either close this bug out or validate the issue. I'll try to squeeze a test in before EoD.

**Comment by sudermanjr** [comment]
@thejosephstevens Any word on this?

If you want to open another Issue outlining what we could do for more validation on CRDs we would be happy to take a look. Might be a good first issue for someone.

**Comment by thejosephstevens** [comment]
Hey all, sorry for the wait on this. I just tried out this and it worked just fine without declaring a namespace for the clusterrolebinding, looks like pebcak. I appreciate the responsiveness from all of you! I'll add an issue for CRD validation.
```
apiVersion: rbacmanager.reactiveops.io/v1beta1
kind: RBACDefinition
metadata:
  name: node-labeler
rbacBindings:
  - name: node-labeler
    subjects:
      - kind: ServiceAccount
        name: node-labeler
        namespace: infra
    clusterRoleBindings:
      - clusterRole: node-labeler
```

**Comment by sudermanjr** [comment]
Thanks for the feedback!
